src/csharp/Intel/Generator/Assembler/CSharp/CSharpAssemblerSyntaxGenerator.cs

```diff
@@ -0,0 +1,1760 @@
+	[Generator(TargetLanguage.CSharp, GeneratorNames.Encoder)]
+	sealed class CSharpAssemblerSyntaxGenerator : AssemblerSyntaxGenerator {
+		readonly GeneratorOptions _generatorOptions;
+		readonly CSharpDocCommentWriter _docWriter;
+
+		public CSharpAssemblerSyntaxGenerator(GeneratorOptions generatorOptions) {
+			Converter = CSharpIdentifierConverter.Create();
+			_docWriter = new CSharpDocCommentWriter(Converter);
+			_generatorOptions = generatorOptions;
+		}
+
+		IdentifierConverter Converter { get; }
+
+		protected override void GenerateRegisters(EnumType registers) {
+			var filename = Path.Combine(CSharpConstants.GetDirectory(_generatorOptions, CSharpConstants.IcedNamespace), "Assembler", "AssemblerRegisters.g.cs");
+			using (var writer = new FileWriter(TargetLanguage.CSharp, FileUtils.OpenWrite(filename))) {
+				writer.WriteFileHeader();
+				writer.WriteLine($"#if {CSharpConstants.EncoderDefine}");
+
+				writer.WriteLine($"namespace {CSharpConstants.IcedNamespace} {{");
+				writer.WriteLine("#pragma warning disable 1591 // Missing XML comment for publicly visible type or member");
+				using (writer.Indent()) {
+					writer.WriteLine("public static partial class AssemblerRegisters {");
+					using (writer.Indent()) {
+						foreach (var register in registers.Values) {
+							if (register.Value == 0) {
+								continue;
+							}
+							var name = register.Name(Converter);
+							// Discard EIP/RIP from assembler registers
+							if (name == "EIP" || name == "RIP") continue;
+							var registerTypeName = $"AssemblerRegister{GetRegisterPostfix((Register)register.Value)}";
+							writer.WriteLine($"public static readonly {registerTypeName} {name.ToLowerInvariant()} = new {registerTypeName}(Register.{name});");
+						}
+					}
+					writer.WriteLine("}");
+				}
+				writer.WriteLine("}");
+				writer.WriteLine("#endif");
+			}
+		}
+
+		protected override void Generate(Dictionary<GroupKey, OpCodeInfoGroup> map, OpCodeInfoGroup[] groups) {
+			GenerateCode(map, groups);
+			GenerateTests(map, groups);
+		}
+		
+		void GenerateCode(Dictionary<GroupKey, OpCodeInfoGroup> map, OpCodeInfoGroup[] groups) {
+			var filename = Path.Combine(CSharpConstants.GetDirectory(_generatorOptions, CSharpConstants.IcedNamespace), "Assembler", "Assembler.g.cs");
+			using (var writer = new FileWriter(TargetLanguage.CSharp, FileUtils.OpenWrite(filename))) {
+				writer.WriteFileHeader();
+				writer.WriteLine($"#if {CSharpConstants.EncoderDefine}");
+				writer.WriteLine($"namespace {CSharpConstants.IcedNamespace} {{");
+				using (writer.Indent()) {
+					writer.WriteLine("using System;");
+					writer.WriteLine("public sealed partial class Assembler {");
+					using (writer.Indent()) {
+						foreach (var group in groups) {
+							var renderArgs = GetRenderArgs(group, out var hasImm8);
+							var methodName = Converter.Method(group.Name);
+							RenderCode(writer, methodName, group, renderArgs);
+							if (hasImm8) {
```

Comment:
`hasImm`? Then byte -> sbyte, int/long -> uint/ulong in `ChangeByteToSignedByteArguments()`